docs/architecture/adr-023-data-fee-market.md

```diff
@@ -0,0 +1,60 @@
+# ADR 023: DA Fee Market
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/06/28: Initial draft
+
+## Introduction
+
+At it's core, the Celestia network can be described as a market for the commodity of published data. By outsourcing the publication of data to Celestia, a rollup can mitigate the possibility of equivocation that may cause different machines to execute on a different set and order of transactions resulting in different states. In the field of distributed systems, this is known as split-brain. Thus, the market encompasses buyers who are willing to pay various amounts on the publication of their data, and sellers who are willing to supply the machines to perform the computation, storage and networking as defined by the protocol. This document explores the various dynamics of this market and makes some preliminary decisions on how to best design the fee market.
+
+## Context
+
+There are several imporant considerations regarding Celestia's fee market model:
+
+- **A Single Decentralized Seller**: While there are multiple competing buyers, the group of "validators" acting as sellers must collectively perform the computation, storage and networking that is required to produce the service. They may incur different costs in running their machines and receive different slices of the overall revenue generated (see non-linear distribution) but nonetheless must perform mostly the equivalent service. More importantly, the quality of the service is proportional to the level of decentralization of the network. More evenly weighted nodes makes the network more resilient to attacks and censorship and improves the reliability to accessing the published data. In summary, the force to continually lower costs to buyers and to remain a competitive market may reduce the amount of sellers, yet counter to this, the security and thus the quality of the service is dependent on a healthy amount of sellers. The market needs to balance this.
```

Comment:
> More evenly weighted nodes makes the network more resilient to attacks and censorship and improves the reliability to accessing the published data.

we might want to distinguish between per block censorship resistance and more medium term censorship reistance. The reason being that having an even distribution of stake doesn't increase per block censorship resistance.